GMMK2 65% ISO: Correct layout data

noroadsleft opened the pull request:
## Description

Correct key sizes/positioning on the right side of the board.

Before:
![image](https://github.com/qmk/qmk_firmware/assets/18669334/73a50338-cb47-4133-8279-8d4948245ab1)

After:
![image](https://github.com/qmk/qmk_firmware/assets/18669334/f8a89126-ab03-4dc8-aad2-c76ed52979c0)

cc @itarze (keyboard maintainer)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist*document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING*document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

dunk2k commented:
Hi @noroadsleft,
I caught this one, corrected ISO Enter matrix sequence and added community layout support in #21855 🙈

noroadsleft commented:
> I caught this one, corrected ISO Enter matrix sequence and added community layout support in `#21855` 🙈

Appreciate the effort and the heads-up.

Not a fan of a 428-file PR, though :( It's why I split my updates to be keyboard-specific (or sometimes vendor-specific).

dunk2k commented:
> Not a fan of a 428-file PR, though :(

The amount of work it's involved, I'll be avoiding big'ns like that in future.

More than happy for you to replicate layout and community layout additions, of this keyboard, from my mentioned PR in this PR so I can knock that 428 file count down a smidge. 🙂
